docs/blockentities/index.md

```diff
@@ -1,141 +1,239 @@
 
 ```java
-// In some subclass of BlockEntity
-@Override
-public CompoundTag getUpdateTag(HolderLookup.Provider registries) {
-  CompoundTag tag = new CompoundTag();
-  //Write your data into the tag
-  return tag;
+// Note: The ticker is defined in the block, not the block entity. However, it is good practice to
+// keep the ticking logic in the block entity in some way, for example by defining a static #tick method.
+public class MyEntityBlock extends Block implements EntityBlock {
+    // other stuff here
+
+    @Override
+    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level level, BlockState state, BlockEntityType<T> type) {
+        // You can return different tickers here, depending on whatever factors you want. A common use case would be
+        // to return different tickers on the client or server, or only tick one side to begin with.
+        return type == MY_BLOCK_ENTITY.get() ? MyBlockEntity::tick : null;
+    }
 }
 
-@Override
-public Packet<ClientGamePacketListener> getUpdatePacket() {
-  // Will get tag from #getUpdateTag
-  return ClientboundBlockEntityDataPacket.create(this);
+public class MyBlockEntity extends BlockEntity {
+    // other stuff here
+
+    // The signature of this method matches the signature of the BlockEntityTicker functional interface.
+    public static void tick(Level level, BlockPos pos, BlockState state, MyBlockEntity blockEntity) {
+        // Whatever you want to do during ticking.
+        // For example, you could change a crafting progress value or consume power here.
+    }
 }
+```
+
+Be aware that the `#tick` method is actually called every tick. Due to this, you should avoid doing a lot of complex calculations in here if you can, for example by only calculating things every X ticks, or by caching the results.
+
+## Syncing
 
-// Can override IBlockEntityExtension#onDataPacket. By default, this will defer to  BlockEntity#loadWithComponents.
+Block entity logic is usually run on the server. As such, we need to tell the client what we are doing. There are three ways to do just that: on chunk load, on block update, or by using a custom packet. You should generally only sync information when it is necessary, to not needlessly clog up the network. 
+
+### Syncing on Chunk Load
+
+A chunk is loaded (and by extension, this method is utilized) each time it is read from either network or disk. To send your data here, you need to override the following methods:
+
+```java
+public class MyBlockEntity extends BlockEntity {
+    // ...
+
+    // Create an update tag here. For block entities with only a few fields, this can just call #saveAdditional.
+    @Override
+    public CompoundTag getUpdateTag(HolderLookup.Provider registries) {
+        CompoundTag tag = new CompoundTag();
+        // You can also opt to only save the data that has actually changed here.
```

Comment:
`getUpdateTag()` must always send all the data the client needs. Omitting unchanged data is only an option in `getUpdatePacket()` when the packet constructor with the `BiFunction<BlockEntity, RegistryAccess, CompoundTag>` parameter is used.